api/rest/restapi_test.go

```diff
@@ -148,6 +155,53 @@ func TestAPIPeerAddEndpoint(t *testing.T) {
 	}
 }
 
+func TestAPIAddFileEndpoint(t *testing.T) {
```

Comment:
Either make 3 tests, or make 3 subtests (`t.Run()`)

Also, I think you need to rebase sharing on master. All the restapi tests are using this `testBothEndpoints` format that runs the test against both the libp2p and the http endpoints, so it should be used here too.